source/client/service_impl.cc

```diff
@@ -0,0 +1,91 @@
+#include "client/service_impl.h"
+
+#include <grpc++/grpc++.h>
+
+#include "client/client.h"
+#include "client/options_impl.h"
+
+namespace Nighthawk {
+namespace Client {
+
+void ServiceImpl::handleExecutionRequest(const nighthawk::client::ExecutionRequest& request) {
+  ASSERT(running_, "running_ ought to be set");
+  nighthawk::client::ExecutionResponse response;
+  response.mutable_error_detail()->set_code(grpc::StatusCode::INTERNAL);
+
+  OptionsPtr options;
+  try {
+    options = std::make_unique<OptionsImpl>(request.start_request().options());
+  } catch (Envoy::EnvoyException exception) {
+    response.mutable_error_detail()->set_message(exception.what());
+    writeResponseAndFinish(response);
+    return;
+  }
+
+  ProcessImpl process(*options, time_system_);
+  OutputCollectorFactoryImpl output_format_factory(time_system_, *options);
+  auto logging_context = std::make_unique<Envoy::Logger::Context>(
+      spdlog::level::from_str(options->verbosity()), "[%T.%f][%t][%L] %v", log_lock_);
+  auto formatter = output_format_factory.create();
+  if (process.run(*formatter)) {
+    response.clear_error_detail();
+    *(response.mutable_output()) = formatter->toProto();
+  } else {
+    response.mutable_error_detail()->set_message("Unknown failure");
```

Comment:
Nit: This line is missing test coverage.

Reply:
Fixed